### ISSUE TITLE
Suboptimal layout for simple case

### ISSUE THREAD

**Issue by qbolec** [cpettitt/dagre#171]
Consider this example:
http://jsfiddle.net/964274dL/1/
with following edges
```
 var edges = [[25,23],[58,31],[20,37],[129,37],[26,50],[31,57],[26,65],[23,129],[158,157],[159,157],[160,158],[160,159],[174,165],[186,182],[242,182],[248,184],[169,188],[255,188],[199,215],[199,224],[255,224],[276,233],[225,241],[248,257],[270,257],[272,271],[271,276]];
```
If you look at the right part of the image, you can see several "suboptimal" features:
- the arrow from 255 to 188 has an unnecessary turn, 
- the arrow from 225 to 241 is a separate component of the graph and thus it does not have to intersect anything, but it does

Am I doing something wrong?
It seems to me that above issues could be solved by swapping neighbouring nodes with each other, which I believe is one of simplest heuristics.

**Comment by eliaslevy** [comment]
Agreed.  For instance, take this simple graph: http://www.tiikoni.com/tis/view/?id=706f8b4

The graph is largely a tree, with a subset of the leaf nodes having multiple parents.  One would expect the majority of the single parent leaf nodes to group themselves around their parent one rank lower, while leaf nodes that have multiple parents would be located somewhere between parents.  This would minimize the number of edge cross.

Instead the crossing minimizing algorithm seems to fail.  For instance, the two top most leaf nodes, f74ef... and d2f448... should obviously be swapped to avoid their crossing.  There are many other unnecessary crossing in the layout.

Obviously a lot of work has gone into dagre, but for some reason these simple graphs are failing to be laid out correctly.
